src/java.base/share/classes/java/util/zip/ZipUtils.java

```diff
@@ -170,23 +171,31 @@ static LocalDateTime javaEpochToLocalDateTime(long time) {
      * The bytes are assumed to be in Intel (little-endian) byte order.
      */
     public static final int get16(byte[] b, int off) {
-        return (b[off] & 0xff) | ((b[off + 1] & 0xff) << 8);
+        Preconditions.checkIndex(off, b.length, Preconditions.AIOOBE_FORMATTER);
+        Preconditions.checkIndex(off + 1, b.length, Preconditions.AIOOBE_FORMATTER);
```

Comment:
Please use `Preconditions.checkFromIndexSize`, which should be less overhead:
```suggestion
        Preconditions.checkFromIndexSize(off, 2, b.length, Preconditions.AIOOBE_FORMATTER);
```
Similarly for other methods.

Reply:
It's actually not less overhead in my tests, since `checkIndex` is intrinsic and mostly disappears, while with checkFromIndexSize the bounds checks remain. It's on my todo to investigate this but I think `checkFromIndexSize` needs to be given similar intrinsification treatment as `checkIndex`.

Reply:
Actually if we trust the input index to be nonnegative, we can just check our end index for out of bounds too.

Reply:
Sure, I think the JIT is pretty good at eliminating the (intrinsic) `checkIndex` calls when they are redundant though, so we don't have to open up that can of worms and can keep these utility methods safe. 

FWIW I wouldn't mind giving similar treatment to `ByteArray(-LittleEndian)` and avoid the `VarHandles` dependency in those utility classes, but I have no urge to get into the sort of discussions that were spawned in #19616

Reply:
> Actually if we trust the input index to be nonnegative, we can just check our end index for out of bounds too.

I would not trust that. Perhaps for well-formed ZIP files, but trust me, not all ZIPs are well-formed ;-)

Reply:
Yep, that requires you to pre-validate the input argument as a fixed position after already-read content, which is not always the case.

Reply:
Like @eirbjo suggests we'd have to put a lot of validation in other places if we went down this route. Regardless this is an academic discussion since the PR suggests the safe route and we don't pay much of a cost for that in microbenchmarks.